packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue

```diff
@@ -60,13 +65,31 @@ const processJoin = async (accept: boolean) => {
     return
   }
 
+  const userId = activeUser.value?.id
+  if (!userId) return
+
   const result = await apollo
     .mutate({
       mutation: DashboardJoinWorkspaceDocument,
       variables: {
         input: {
           workspaceId: props.workspace.id
         }
+      },
+      update(cache) {
+        modifyObjectField(
+          cache,
+          getCacheId('User', userId),
+          'workspaces',
+          ({ helpers: { createUpdatedValue, ref } }) => {
+            return createUpdatedValue(({ update }) => {
```

Comment:
@cdriesler User.workspaces has variables as well, you usually want to just evict all filtered (e.g. search specified) cache values cause there's no way to accurately tell if the new workspace fits that filter or not, and then make the cache update to the list only in non-filtered values

Reply:
Evict means - wherever that evicted value was referenced, e.g. in some kind of query living in a UI component, the query will understand that its now missing a part of its data and refetch it